Jetcaster/app/src/main/java/com/example/jetcaster/ui/MainActivity.kt

```diff
@@ -30,10 +40,31 @@ class MainActivity : ComponentActivity() {
         // This app draws behind the system bars, so we want to handle fitting system windows
         WindowCompat.setDecorFitsSystemWindows(window, false)
 
+        /**
+         * Flow of [DevicePosture] that emits every time there's a change in the windowLayoutInfo
+         */
+        val devicePosture = windowInfoRepository().windowLayoutInfo
+            .flowWithLifecycle(this.lifecycle)
+            .map { layoutInfo ->
+                val foldingFeature =
+                    layoutInfo.displayFeatures.filterIsInstance<FoldingFeature>().firstOrNull()
+                when {
+                    isTableTopPosture(foldingFeature) ->
+                        DevicePosture.TableTopPosture(foldingFeature?.bounds!!)
```

Comment:
This might be too advanced, but a contract would avoid the need for a potentially unsafe `!!` here:

```
@OptIn(ExperimentalContracts::class)
fun isTableTopPosture(foldFeature: FoldingFeature?): Boolean {
    contract { returns(true) implies (foldFeature != null) }
    return foldFeature?.state == FoldingFeature.State.HALF_OPENED &&
        foldFeature.orientation == FoldingFeature.Orientation.HORIZONTAL
}
```

Reply:
Neat! Thank you :) added